shuttle_test.go

```diff
@@ -276,20 +334,19 @@ func BenchmarkPipeline(b *testing.B) {
 	config.LogsURL = ts.URL
 	config.InputFormat = InputFormatRaw
 
-	b.ResetTimer()
+	shut := NewShuttle(config)
+	input := NewLoopingBuffer(longerTestData)
+	shut.LoadReader(input)
+	shut.Launch()
+	var tb int
 	for i := 0; i < b.N; i++ {
-		b.StopTimer()
-		shut := NewShuttle(config)
-		input := NewLongerTestInput()
-		b.SetBytes(int64(input.Len()))
-
-		shut.LoadReader(input)
-
-		b.StartTimer()
-		shut.Launch()
-		shut.WaitForReadersToFinish()
-		shut.Land()
+		// This sleep is here to allow the reading goroutines to make
```

Comment:
I not only have to wait, but also need the loop so bytes get shoved through it. I played with all sorts of ways to do this before punting on a sleep. I need to look more at what net/http is doing with that hook probably.